### PR TITLE
feat: Add experimental visionOS support

### LINKED ISSUE #3101: Add Vision OS Sample
### Description

Once the [VisionOS SDK](https://developer.apple.com/visionos/) is available, we should add a simple sample application or test one of our sample apps with the VisionOS SDK for compatibility.

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Added support for visionOS

## :bulb: Motivation and Context

close #3101, #3323 

## :green_heart: How did you test it?

Samples

## :pencil: Checklist

You have to check all boxes before merging:

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by philipphofmann — COMMENTED** [review]
Awesome, I think we should also add validate visionOS with pod lib lint https://github.com/getsentry/sentry-cocoa/blob/05644fec828168911b8a2177b583ca6cb571b731/.github/workflows/lint.yml#L45-L52 

and we should also build the sample in CI similar to what we do for watchOS https://github.com/getsentry/sentry-cocoa/blob/05644fec828168911b8a2177b583ca6cb571b731/.github/workflows/build.yml#L70-L76

**Comment by brustolin** [comment]
> Awesome, I think we should also validate visionOS with pod lib lint

Right not is not possible because GitHub don't support Xcode 15.1 yet, should we leave this PR open until GitHub adds support, or should we release it with this experimental feature so devs that are also experimenting visionOS could try Sentry with it?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Features
 
+- Add visionOS support (#3328)
```
Comment (philipphofmann) [inline CHANGELOG.md:7]:
```suggestion
- Add experimental visionOS support (#3328)
```

**Review by philipphofmann — APPROVED** [review]
LGTM

**Comment by philipphofmann** [comment]
@brustolin, can we merge this?

**Comment by philipphofmann** [comment]
@brustolin, the `Test / UI Tests for ios_swift on Simulators (pull_request) ` is failing on main since we merged your PR https://github.com/getsentry/sentry-cocoa/pull/3298. I'm having a look why it's broken.

**Comment by philipphofmann** [comment]
@brustolin, I reverted the PR with https://github.com/getsentry/sentry-cocoa/pull/3370 to have a stable main branch again.

File: Sentry.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -6159,8 +6172,11 @@
 				MTL_FAST_MATH = YES;
 				PRODUCT_BUNDLE_IDENTIFIER = io.sentry.SentryPrivate;
 				PROVISIONING_PROFILE_SPECIFIER = "";
+				SUPPORTED_PLATFORMS = "appletvos appletvsimulator iphoneos iphonesimulator macosx visionos watchos watchsimulator xros xrsimulator";
+				SUPPORTS_MACCATALYST = YES;
 				SWIFT_EMIT_LOC_STRINGS = YES;
 				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2,3,4,7";
```
Comment (armcknight) [inline Sentry.xcodeproj/project.pbxproj:6179]: These are already all set in SDK.xcconfig, so I'll push a commit to remove them.

**Comment by philipphofmann** [comment]
If I'm not mistaken, we can merge this @brustolin?

**Pull request merged**